src/client/components/equalizer/equalizer.js

```diff
@@ -0,0 +1,143 @@
+
+
+class Equalizer extends Component {  
+   componentDidMount(){ 
+      this.detectStreamSoundFromMicrophon();
+    }
+  
+  createSoundStream = (stream) => {
+    const { audioContext: context } =  this.props.audioData;           
+    const audioLineIn = new Audio(); 
+    
+    audioLineIn.srcObject = stream;
+    audioLineIn.muted=true;        
+    //hark - JS module that listens to an audio stream, and emits events indicating whether the user is speaking or not 
+    const options = {};
+    const speechEvents = Hark(stream, options);
+    const htmlinfo=document.getElementById("stream_detecting");      
+    speechEvents.on('speaking', ()=>{           
+            htmlinfo.innerHTML=`speaking`;
+          });
+       
+    speechEvents.on('stopped_speaking', ()=>{
+            htmlinfo.innerHTML=`no stream detekting`;
+          });
+    const sourceStream = context.createMediaStreamSource(stream);                           
+    this.props.createStreamData({audioLineIn: audioLineIn, sourceStream});
+  }  
+
+  detectStreamSoundFromMicrophon = () => {
+    if (navigator.mediaDevices) {      
+      navigator.mediaDevices.getUserMedia ({audio: true})      
+      .then(stream => this.createSoundStream(stream))
+      .catch((e) => {
+        throw new Error(e);
+      });
+  } else {
+    throw new Error('browser doesnt support audio API');
+  }  
+  }  
+
+  widthMerge = (e) => {   
+    this.props.mergeCanvasWidth(e);
+  }     
+
+  playSoundFromFile = () => {
+    const {audioFromFile: soundfromfile, playPauseState} = this.props.audioData;    
+    if (!playPauseState) {     
+      soundfromfile.play();
+      this.equaliserRun();     
+    } else {      
+      soundfromfile.pause();   
+  }
+    this.props.playPauseSoundFromFile();
+  }
+
+  startMuteStream = () => {    
+    const {audioStream, audioContext, analyser, streamSource, startMuteState} = this.props.audioData; 
+                 
+    if (!startMuteState) { 
+      streamSource.connect(analyser);
+      analyser.connect(audioContext.destination); 
+      //play/pause function doesn't work 
+      audioStream.play();
+      this.equaliserRun();
+      this.props.startMuteStreamAudio();
+    } else { 
+      streamSource.disconnect(analyser);
+      audioStream.pause();
+      this.props.startMuteStreamAudio();
+  }
+  }
+
+  equaliserRun = () => {   
+    const ctx = document.querySelector("canvas").getContext("2d");    
+    const {analyser, audioContext} = this.props.audioData;    
+    const numPoints = analyser.frequencyBinCount-80;
+    const heightArray = new Uint8Array(numPoints); 
+    const equalizerMethods = new EqualizerMethods(ctx, audioContext, analyser, numPoints, heightArray);   
+    requestAnimationFrame(equalizerMethods.renderEqualizer);
+  }   
+   
+
+  uploadSoundInfoFromFile = (e) => {  
+    const file=e.target.files[0];
```

Comment:
Could be destructed ‘const [file] = e.target.files;

Reply:
ok